net-im/libcommuni/libcommuni-3.5.0.ebuild

```diff
@@ -0,0 +1,73 @@
+# Copyright 1999-2018 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=6
+
+inherit qmake-utils
+
+DESCRIPTION="A cross-platform IRC framework written with Qt"
+HOMEPAGE="http://communi.github.io/"
+SRC_URI="https://github.com/communi/libcommuni/archive/v${PV}.tar.gz -> ${P}.tar.gz"
+
+LICENSE="BSD"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE="qml test +uchardet"
+
+# Compiling with USE="test" is currently broken with dev-libs/icu
+# See: https://github.com/communi/libcommuni/issues/60
+REQUIRED_USE="test? ( uchardet )"
+
+RDEPEND="dev-qt/qtcore:5
+	dev-qt/qtdeclarative:5
+	uchardet? ( app-i18n/uchardet )
+	!uchardet? ( dev-libs/icu:= )"
+
+DEPEND="test? ( dev-qt/qttest:5 )
+	${RDEPEND}"
+
+RESTRICT="!test? ( test )"
+
+src_prepare() {
+	default
+
+	# Don't use bundled uchardet, use instead app-i18n/uchardet
+	rm -rf src/3rdparty/uchardet-0.0.1/* || die
+	cp "${FILESDIR}"/uchardet.pri src/3rdparty/uchardet-0.0.1/uchardet.pri || die
+}
+
+src_configure() {
+	local myqmakeargs=( libcommuni.pro
+		# Disables compile run-time benchmarks, as they don't make any sense
+		-config no_benchmarks
+		# Disables examples, as no new files are installed
+		-config no_examples
+		# Is needed for fixing the QA Notice: The following files contain insecure RUNPATHs
+		-config no_rpath
+		# Compile libcommuni always in Release mode, as Debug seems to do nothing
+		-config release
+		# Don't silence all compile messages
+		-config verbose
+		-config $(usex qml '' 'no_')install_imports
+		-config $(usex qml '' 'no_')install_qml
+		-config $(usex test '' 'no_')tests
+		-config $(usex uchardet 'no_' '')icu
+		-config $(usex uchardet '' 'no_')uchardet )
+
+	eqmake5 "${myqmakeargs[@]}"
+}
+
+src_test() {
+	# This is an ugly hack to modify LD_LIBRARY_PATH, otherwise a test will fail,
+	# because it uses the system installed lib instead of the new compiled lib.
+	# The test will fail, when libcommuni is emerged with USE="uchardet" and
+	# libcommuni is already installed with USE="-uchardat", or the other way around.
+	find "${S}" -type f -name 'target_wrapper.sh' -exec sed -i -e "s/\/usr\/lib64\${LD_LIBRARY_PATH:+:\$LD_LIBRARY_PATH}/${S//\//\\/}\/lib/" {} \; || die
```

Comment:
@gentoo/qt, could you advise here? Is there a better way to make qmake behave sanely?

Reply:
Define "behave sanely"... Are you talking about setting an rpath on the test executables? In that case you can use `QMAKE_RPATHDIR`. Other than that, qmake doesn't do anything special on its own AFAIK.

Reply:
The generated `target_wrapper.sh` includes ` LD_LIBRARY_PATH=/usr/lib64${LD_LIBRARY_PATH:+:$LD_LIBRARY_PATH}`

Reply:
This sed really looks fragile. Instead, please remove LD_LIBRARY_PATH from that wrapper, and set it in ebuild.

Reply:
Done.

Reply:
I menat: `d` the whole line from that file, and set `local -x LD_LIBRARY_PATH=...` inside ebuild. Passing paths to sed expressions is a really bad idea.

Reply:
Okay, changed.